include/grpcpp/impl/codegen/client_callback.h

```diff
@@ -594,7 +594,7 @@ class ClientCallbackReaderWriterImpl
     start_tag_.Set(
         call_.call(),
         [this](bool ok) {
-          reactor_->OnReadInitialMetadataDone(ok);
+          reactor_->OnReadInitialMetadataDone(ok && !context_->trailers_only());
```

Comment:
Do we need to also make this same change for `OnReadDone()`?  Or does it already get ok=false when there is no message to read?

Reply:
Yeah, OnReadDone already gets ok=false if there's no message available.